src/services/storage/storage.service.ts

```diff
@@ -264,8 +325,28 @@ export class Storage {
    * Migrates existing data to match the current version
    * @returns promise
    */
-  public migrate(): Promise<void> {
-    return migrate()
+  public async migrate(newVersionString: string): Promise<void> {
+    const currentVersionString = await this.getStorageVersion()
+    let currentVersion: Version
+
+    try {
+      currentVersion = versionFromString(currentVersionString)
+    } catch (error) {
+      currentVersion = {
+        major: 0,
+        minor: 0,
+        patch: 0,
+      }
+    }
+
+    const accountDapps = await getObject<AccountDapps>(Storage.dappsKey, accountDappsFactory)
+    const updatedAccountDapps = migrateDapps(accountDapps, currentVersion)
+
+    if (updatedAccountDapps) {
+      await updateObject<AccountDapps>(Storage.dappsKey, updatedAccountDapps)
+    }
```

Comment:
as I see it does not upgrade anything now but these migrations could extracted into separate migration functions somewhere else, because the storage.service is really heavy already.